src/mongo/db/storage/tokuft/tokuft_dictionary.cpp

```diff
@@ -80,13 +81,13 @@ namespace mongo {
         }
 
         int _getWriteFlags(OperationContext *opCtx, bool skipPessimisticLocking=false) {
-            return (skipPessimisticLocking || _isReplicaSetSecondary(opCtx))
+            return (skipPessimisticLocking || _isReplicaSetSecondary(opCtx) || opCtx->HasBounds())
                     ? DB_PRELOCKED_WRITE
                     : 0;
         }
 
         int _getReadFlags(OperationContext *opCtx, bool skipPessimisticLocking) {
-            return (skipPessimisticLocking || _getDBTxn(opCtx).is_read_only() || _isReplicaSetSecondary(opCtx))
+            return (skipPessimisticLocking || _getDBTxn(opCtx).is_read_only() || _isReplicaSetSecondary(opCtx) || opCtx->HasBounds())
```

Comment:
Some FT operations require the DB_PRELOCKED flags passed in because those operations don't need lock tree locks.  This includes read only MongoDB transactions.  This also includes ops performed on write transactions that already have the rows locked (i.e. no need to lock them again).

Similarly, for FT SE cursor creation, we want to skip pre-locking upfront if we know the operation is read only. So `OperationShouldPrelockCursor` tells us that.  If `true`, we will create a cursor that will pre-lock for its write operations. If `false`, we create a cursor that will _not_ pre-lock for its read operations.  The future read-only cursor non-pre-locked read operations will use the correct read/write flags later on, via _getReadFlags.